### PR TITLE
Add a watchdog to detect stalls in the Enttec Widget.

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: common/io/KQueuePoller.cpp [hunk 1]
```diff
@@ -392,7 +392,7 @@ bool KQueuePoller::ApplyChange(int fd, int16_t filter, uint16_t flags,
     int r = kevent(m_kqueue_fd, m_change_set, m_next_change_entry, NULL, 0,
                    NULL);
     if (r < 0) {
-      OLA_WARN << "Failed to apply kqueue changes";
+      OLA_WARN << "Failed to apply kqueue changes:" << strerror(errno);
```
Comment (peternewman) [inline common/io/KQueuePoller.cpp]: SPaG, add a space after the colon.

File: common/utils/Watchdog.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  m_enabled = false;
+  m_fired = false;
+}
+
+void Watchdog::Kick() {
+  MutexLocker lock(&m_mu);
+  m_count = 0;
+}
+
+void Watchdog::Clock() {
```
Comment (peternewman) [inline common/utils/Watchdog.cpp]: Aside from the similarity to Kick, would Tick be a better name than Clock, given it's an action?

File: include/ola/util/Watchdog.h [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+   * @brief Kick the watchdog to avoid a reset.
+   *
+   * This should be called regularly while the watchdog is enabled.
+   */
+  void Kick();
+
+  /**
+   * @brief Check if the process has stalled due to a lack of Kick() calls.
+   *
+   * This should be called regularly. If it's been
```
Comment (peternewman) [inline include/ola/util/Watchdog.h]: Missing words.

**Comment by peternewman** [comment]
Just a few comments.

File: common/utils/Watchdog.cpp (hunk 2 above)
Comment (nomis52) [inline common/utils/Watchdog.cpp]: I took the names from  https://en.wikipedia.org/wiki/Watchdog_timer

Comment (peternewman) [inline common/utils/Watchdog.cpp]: I guessed that may be the case, I'd still argue the input line is the clock, but it ticks.

**Comment by peternewman** [comment]
I'd still prefer the function to be renamed (although I can see where you're coming from), but other than that LGTM

**Pull request merged**